thrift.js

```diff
@@ -234,42 +241,34 @@ Thrift.prototype.compileInclude = function compileInclude(def) {
 
 Thrift.prototype.compileStruct = function compileStruct(def) {
     var self = this;
-    var spec = new ThriftStruct({strict: self.strict});
-    spec.compile(def, self);
-    self.define(spec.fullName, def, spec);
-    self.structs[spec.fullName] = spec;
-    self.types[spec.fullName] = spec;
-    return spec;
+    var model = new ThriftStruct({strict: self.strict});
+    model.compile(def, self);
+    self.define(model.fullName, def, model);
+    return model;
 };
 
 Thrift.prototype.compileException = function compileException(def) {
     var self = this;
-    var spec = new ThriftStruct({strict: self.strict});
-    spec.compile(def, self);
-    self.define(spec.fullName, def, spec);
-    self.exceptions[spec.fullName] = spec;
-    self.types[spec.fullName] = spec;
-    return spec;
+    var model = new ThriftStruct({strict: self.strict, isException: true});
+    model.compile(def, self);
+    self.define(model.fullName, def, model);
+    return model;
 };
 
 Thrift.prototype.compileUnion = function compileUnion(def) {
     var self = this;
-    var spec = new ThriftUnion({strict: self.strict});
-    spec.compile(def, self);
-    self.define(spec.fullName, def, spec);
-    self.unions[spec.fullName] = spec;
-    self.types[spec.fullName] = spec;
-    return spec;
+    var model = new ThriftUnion({strict: self.strict});
+    model.compile(def, self);
+    self.define(model.fullName, def, model);
+    return model;
 };
 
 Thrift.prototype.compileTypedef = function compileTypedef(def) {
     var self = this;
-    var spec = new ThriftTypedef();
-    spec.compile(def, self);
-    self.define(spec.name, spec, spec);
-    self.typedefs[spec.name] = spec;
-    self.types[spec.name] = spec;
-    return spec;
+    var model = new ThriftTypedef({strict: self.strict});
+    model.compile(def, self);
+    self.define(model.name, def, model);
+    return model;
```

Comment:
These functions are starting to look homogenous and can likely be collapsed into _compile eventually.